### PR TITLE
fix: Update coordinator.py to address a startup race condition

### PR DESCRIPTION
**Pull request by caelor**
Fix for https://github.com/fuatakgun/eufy_security/issues/252

### CODE REVIEW

File: custom_components/eufy_security/coordinator.py [hunk 1]
```diff
@@ -181,6 +183,10 @@ async def async_start_listening(self):
         return await wait_for_value(self.__dict__, "devices", None)
```
Comment (fuatakgun) [inline custom_components/eufy_security/coordinator.py:183]: this line should guarantee that `self.devices is not None`

File: custom_components/eufy_security/coordinator.py [hunk 2]
```diff
@@ -164,6 +164,8 @@ async def set_devices(self):
             _LOGGER.debug(
                 f"{DOMAIN} - connect - async_start_listening False - {self.config.__dict__}"
             )
+            # close down the websocket to fully retry the connection
+            await self.ws.ws.close()
```
Comment (fuatakgun) [inline custom_components/eufy_security/coordinator.py:168]: have you checked this against captcha condition? I was intentionally leaving the connection open as to enter captcha, add-on (web socket) should keep the internal state.

File: custom_components/eufy_security/coordinator.py (hunk 1 above)
Comment (caelor) [reply custom_components/eufy_security/coordinator.py:183]: That isn't in the code path that gets executed - as far as I can tell, async_start_listening is only called from set_captcha_if_required_and_user_input if the captcha is needed. That's not the execution path that's been followed for the issue I've seen.

File: custom_components/eufy_security/coordinator.py (hunk 2 above)
Comment (caelor) [reply custom_components/eufy_security/coordinator.py:168]: I've not checked that path. I've never yet been prompted for a captcha, and I could see how closing the websocket while going through the captcha process. Adding a check for `self.captcha_config.required == False` may be prudent (if I understand the code correctly).

File: custom_components/eufy_security/coordinator.py (hunk 1 above)
Comment (fuatakgun) [reply custom_components/eufy_security/coordinator.py:183]: I had forgot to publish these comments from 3 days ago...

Comment (caelor) [reply custom_components/eufy_security/coordinator.py:183]: I think that would definitely help, on the assumption that if the integration connects to the addon at the specific point that is triggering this race, then it correctly sends the push message onwards (which it should). My only concern is my initial attempt to workaround the race was where I didn't tear down the websocket, and just let the integration raise an exception. This caused HomeAssistant to retry setting up the integration every now and then, but left the websocket open. When I tried that approach, we never received the devices message from the addon. Tearing down and re-establishing the websocket after the push connection was established seemed to initialise properly.

Of course, if the addon isn't sending the devices message when it should, that would be an upstream bug.

Comment (caelor) [reply custom_components/eufy_security/coordinator.py:183]: Ok, confirmed still affected with published integration 2.2.4, addon 0.8.4, home assistant 2022.2.6 and OS 7.4:

```
2022-02-13 08:34:08.077  INFO  Eufy Security server listening on host 0.0.0.0, port 3000 
2022-02-13 08:34:11.194  INFO  Connected to station T8010N********** on host 192.168.#.# and port 15355 
2022-02-13 08:34:20.571  INFO  Push notification connection successfully established 
```

and

```
2022-02-13 08:34:13 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry localhost for eufy_security
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 327, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/config/custom_components/eufy_security/__init__.py", line 51, in async_setup_entry
    await coordinator.initialize()
  File "/config/custom_components/eufy_security/coordinator.py", line 87, in initialize
    await self.set_devices()
  File "/config/custom_components/eufy_security/coordinator.py", line 163, in set_devices
    if await self.async_get_device_properties() is False:
  File "/config/custom_components/eufy_security/coordinator.py", line 184, in async_get_device_properties
    for device in self.devices.values():
AttributeError: 'NoneType' object has no attribute 'values'
```
and the integration moves to "Failed to set up" in the UI. The connection to the addon happened at 08:34:13 - after the addon connected to the station but before the push notification was established.

So the "good" news is that fully up to date still exposes the race condition.

Adding in the proposal to Line 183:

```
2022-02-13 08:43:30.879  INFO  Eufy Security server listening on host 0.0.0.0, port 3000 
2022-02-13 08:43:33.367  INFO  Connected to station T8010N********** on host 192.168.#.# and port 23823 
2022-02-13 08:43:43.303  INFO  Push notification connection successfully established 
```
and
```
2022-02-13 08:43:47 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry localhost for eufy_security
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 327, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/config/custom_components/eufy_security/__init__.py", line 51, in async_setup_entry
    await coordinator.initialize()
  File "/config/custom_components/eufy_security/coordinator.py", line 87, in initialize
    await self.set_devices()
  File "/config/custom_components/eufy_security/coordinator.py", line 164, in set_devices
    if await self.async_get_device_properties() is False:
  File "/config/custom_components/eufy_security/coordinator.py", line 185, in async_get_device_properties
    for device in self.devices.values():
AttributeError: 'NoneType' object has no attribute 'values'
```
The integration moves to "failed to set up" in the integrations UI.


It looks like `wait_for_value` waited for the full 50 * 0.25 = 12.5sec, which has placed the connection after the Push Notification connection has been established, but we're still seeing the same effect - so I guess the Push Notification link was a red herring, and that there is a race condition with something not logged.

What I've concluded from this: 12.5sec delay _isn't_ enough to push us past the race, whereas that ~60sec delay that the HomeAssistant retry causes _is_ enough.

**Pull request merged**